AdminWebsite/AdminWebsite/Security/UserIdentity.cs

```diff
@@ -1,78 +1,56 @@
-﻿using AdminWebsite.Services;
+﻿using AdminWebsite.Helper;
 using System.Collections.Generic;
 using System.Linq;
 using System.Security.Claims;
 
 namespace AdminWebsite.Security
 {
-    /// <summary>
-    /// Active directory information for current user
-    /// </summary>
     public interface IUserIdentity
     {
         IEnumerable<string> GetGroupDisplayNames();
-
         bool IsAdministratorRole();
-
         string GetUserIdentityName();
-
         bool IsVhOfficerAdministratorRole();
-
         bool IsCaseAdministratorRole();
-
-        /// <summary>
-        /// Returns a list of the case types the user is allowed to administrate
-        /// </summary>
         IEnumerable<string> GetAdministratorCaseTypes();
     }
 
     public class UserIdentity : IUserIdentity
     {
-        private static readonly string[] AcceptedAdministratorRoles = { "Civil Money Claims", "Tax", "Financial Remedy" };
-        private static readonly string virtualRoomAdministrator = "VirtualRoomAdministrator";
-
+        private static readonly string[] AcceptedAdministratorRoles = { "MoneyClaims", "FinancialRemedy" };
         private readonly ClaimsPrincipal _currentUser;
-        private readonly IUserAccountService _userAccountService;
+        private readonly AdministratorRoleClaimsHelper _administratorRoleClaimsHelper;
 
-        public UserIdentity(ClaimsPrincipal currentUser, IUserAccountService userAccountService)
+        public UserIdentity(ClaimsPrincipal currentUser)
         {
             _currentUser = currentUser;
-            _userAccountService = userAccountService;
+            _administratorRoleClaimsHelper = new AdministratorRoleClaimsHelper(_currentUser.Claims);
         }
 
         public IEnumerable<string> GetGroupDisplayNames()
         {
-            var groupClaims = _currentUser.Claims.Where(x => x.Type == "groups").ToList();
-            return groupClaims.Select(x => _userAccountService.GetGroupById(x.Value).DisplayName).ToList();            
+            return _administratorRoleClaimsHelper.UserCaseTypes;
         }
 
         /// <inheritdoc />
         public IEnumerable<string> GetAdministratorCaseTypes()
         {
-            return GetGroupDisplayNames().Where(group => AcceptedAdministratorRoles.Contains(group));
+            return _administratorRoleClaimsHelper.UserCaseTypes.Where(group => AcceptedAdministratorRoles.Contains(group));
```

Comment:
Using the `UserCaseTypes` you don't need to compare it with a list of roles anymore, the list will be the types.

Reply:
Good catch!!